apps/dashboard/src/hooks/use-billing-subscription.ts

```diff
@@ -0,0 +1,61 @@
+import { useQuery } from '@tanstack/react-query';
+import { useMemo } from 'react';
+import { differenceInDays, isSameDay } from 'date-fns';
+import { ApiServiceLevelEnum, GetSubscriptionDto } from '@novu/shared';
+import { useAuth } from '@/context';
+import { getBillingSubscription } from '@/api/billing';
+import { QueryKeys } from '@/utils/query-keys';
+
+const today = new Date();
+
+export type UseSubscriptionType = GetSubscriptionDto & { trial: { daysLeft: number }; isLoading: boolean };
+
+export const useBillingSubscription = () => {
+  const { currentOrganization } = useAuth();
+
+  const { data: subscription, isLoading: isLoadingSubscription } = useQuery<GetSubscriptionDto>(
+    [QueryKeys.billingSubscription, currentOrganization?._id],
+    getBillingSubscription,
+    {
+      enabled: !!currentOrganization,
+      initialData: {
+        apiServiceLevel: ApiServiceLevelEnum.FREE,
+        isActive: false,
+        hasPaymentMethod: false,
+        status: 'trialing',
+        currentPeriodStart: null,
+        currentPeriodEnd: null,
+        billingInterval: null,
+        events: {
+          current: 0,
+          included: 0,
+        },
+        trial: {
+          isActive: false,
+          start: today.toISOString(),
+          end: today.toISOString(),
+          daysTotal: 0,
+        },
+      },
+    }
+  );
+
+  const daysLeft = useMemo(() => {
```

Comment:
Nit-picky, we can add this to the API.

Reply:
yep, but that will require changes in three apps: api, web, dashboard

Reply:
We can add only one new field for the Dashboard.